### PR TITLE
[YUNIKORN-1982] Enable manual triggering for GitHub CI

### PR DESCRIPTION
**Pull request by doupache**
### What is this PR for?
Allow running manually from the github ui. 
Everyone can manually run the CI on their fork.

github docs 
1. https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow
2. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

This will not affect **ANY** github action we are running.
### What type of PR is it?
* [x] - Improvement

### What is the Jira issue?
 [YUNIKORN-1982](https://issues.apache.org/jira/browse/YUNIKORN-1982)

### CODE REVIEW

File: .github/workflows/auto-publish.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ on:
   push:
     branches:
       - master
+  workflow_dispatch: {}
```
Comment (wilfred-s) [inline .github/workflows/auto-publish.yml:22]: We do not want the push flow to be run manually.
If something is that broken that a publishing action to the production site needs to be run manually then we need to run it fully manually.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

**Comment by doupache** [comment]
@wilfred-s  
We only manual trigger pre-commit hook. 
Please Take a look. 🙏

**Comment by pbacsko** [comment]
+1 LGTM

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**